experimental/features.py

```diff
@@ -0,0 +1,371 @@
+                 b_std: float = 0.,
+                 channel_axis: int = -1):
+
+  if b_std != 0.0:
+    raise NotImplementedError('Non-zero b_std is not implemented yet .'
+                              ' Please set b_std to be `0`.')
+
+  def init_fn(rng, input_shape):
+    nngp_feat_shape, ntk_feat_shape = input_shape[0], input_shape[1]
+    new_nngp_feat_shape = nngp_feat_shape[:-1] + (nngp_feat_shape[-1] *
+                                                  filter_size**2,)
+    new_ntk_feat_shape = nngp_feat_shape[:-1] + (
+        (nngp_feat_shape[-1] + ntk_feat_shape[-1]) * filter_size**2,)
+    return (new_nngp_feat_shape, new_ntk_feat_shape), ()
+
+  def feature_fn(f, input, **kwargs):
+    nngp_feat, ntk_feat = f.nngp_feat, f.ntk_feat
+
+    nngp_feat = _conv2d_feat(nngp_feat, filter_size) / filter_size * W_std
+
+    if _is_defaut_feature(ntk_feat):  # check if ntk_feat is empty
+      ntk_feat = nngp_feat
+    else:
+      ntk_feat = _conv2d_feat(ntk_feat, filter_size) / filter_size * W_std
+      ntk_feat = np.concatenate((ntk_feat, nngp_feat), axis=channel_axis)
+
+    return f.replace(nngp_feat=nngp_feat, ntk_feat=ntk_feat)
+
+  return init_fn, feature_fn
+
+
+@layer
+def AvgPoolFeatures(window_size: int,
+                    stride_size: int = 2,
+                    padding: str = stax.Padding.VALID.name,
+                    normalize_edges: bool = False,
+                    batch_axis: int = 0,
+                    channel_axis: int = -1):
+
+  def init_fn(rng, input_shape):
+    nngp_feat_shape, ntk_feat_shape = input_shape[0], input_shape[1]
+
+    new_nngp_feat_shape = nngp_feat_shape[:1] + (
+        nngp_feat_shape[1] // window_size,
+        nngp_feat_shape[2] // window_size) + nngp_feat_shape[-1:]
+    new_ntk_feat_shape = ntk_feat_shape[:1] + (
+        ntk_feat_shape[1] // window_size,
+        ntk_feat_shape[2] // window_size) + ntk_feat_shape[-1:]
+    return (new_nngp_feat_shape, new_ntk_feat_shape), ()
+
+  def feature_fn(f, input=None, **kwargs):
+    window_shape_kernel = (1,) + (window_size,) * 2 + (1,)
+    strides_kernel = (1,) + (window_size,) * 2 + (1,)
+    pooling = lambda x: _pool_kernel(x, Pooling.AVG,
+                                     window_shape_kernel, strides_kernel,
+                                     Padding(padding), normalize_edges, 0)
+    nngp_feat = pooling(f.nngp_feat)
+    ntk_feat = pooling(f.ntk_feat)
+
+    return f.replace(nngp_feat=nngp_feat, ntk_feat=ntk_feat)
+
+  return init_fn, feature_fn
+
+
+@layer
+def FlattenFeatures(batch_axis: int = 0, batch_axis_out: int = 0):
+
+  def init_fn(rng, input_shape):
+    nngp_feat_shape, ntk_feat_shape = input_shape[0], input_shape[1]
+    new_nngp_feat_shape = nngp_feat_shape[:1] + (_prod(nngp_feat_shape[1:]),)
+    new_ntk_feat_shape = ntk_feat_shape[:1] + (_prod(ntk_feat_shape[1:]),)
+    return (new_nngp_feat_shape, new_ntk_feat_shape), ()
+
+  def feature_fn(f, input=None, **kwargs):
+    batch_size = f.nngp_feat.shape[0]
```

Comment:
Here and elsewhere, the code may need to be adapted to support different `batch_axis` and `channel_axis`. Feel free to leave it for later.